src/gui/editors/AutomationEditor.cpp

```diff
@@ -2102,5 +2306,25 @@ void AutomationEditorWindow::updateWindowTitle()
 	setWindowTitle( tr( "Automation Editor - %1" ).arg( m_editor->m_clip->name() ) );
 }
 
+void AutomationEditorWindow::setProgressionType(int progType)
+{
+	m_editor->setProgressionType(progType);
+	updateEditTanButton();
+}
+
+void AutomationEditorWindow::updateEditTanButton()
+{
+	auto progType = currentClip()->progressionType();
+
+	if (!AutomationClip::supportsTangentEditing(progType))
+	{
+		if (m_editTanAction->isChecked()) { m_drawAction->trigger(); }
+		if (m_editTanAction->isEnabled()) { m_editTanAction->setEnabled(false); }
+	}
+	else
+	{
+		if (!m_editTanAction->isEnabled()) { m_editTanAction->setEnabled(true); }
+	}
```

Comment:
Maybe this might be simpler (not tested):
```suggestion
	auto progType = currentClip()->progressionType();
	m_editTanAction->setEnabled(AutomationClip::supportsTangentEditing(progType));
	if (!m_editTanAction->isEnabled() && m_editTanAction->isChecked()) { m_drawAction->trigger(); }
```

Reply:
It seems to work fine after a quick test here! Pushing the change (: